### PR TITLE
added find auto interval

### PR DESCRIPTION
**Pull request by ps48**
### Description
The auto interval was hard-coded to span of hours. 

### Issues Resolved
1. Better estimation of Auto interval time frame based on datepicker selection. 
2. Solves selection issue(selecting 'hours' flipped to 'Auto') on Interval selector. 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 1]
```diff
@@ -547,94 +533,118 @@ export const Explorer = ({
...
+            setToast(
+              `Cannot add Visualization '${selectedPanelNameRef.current}' to operation panels, error: ${error.message}`,
+              'danger'
+            );
+          });
       }
     }
   };
 
-  const dateRange = isEmpty(query['selectedDateRange']) ? ['now/15m', 'now'] :
```
[the remark this answers is not in the document]
Comment (ps48) [reply dashboards-observability/public/components/explorer/explorer.tsx:609]: 👍🏽

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 2]
```diff
@@ -108,49 +92,69 @@ export const Explorer = ({
...
+    // less than 2 hours
+    else if (diffSeconds <= 3600 * 2) minInterval = 'm';
+    // less than 2 days
+    else if (diffSeconds <= 86400 * 2) minInterval = 'h';
+    // less than 1 month
+    else if (diffSeconds <= 86400 * 31) minInterval = 'd';
+    // less than 3 months
+    else if (diffSeconds <= 86400 * 93) minInterval = 'w';
+    // less than 1 year
+    else if (diffSeconds <= 86400 * 366) minInterval = 'M';
```
Comment (joshuali925) [inline dashboards-observability/public/components/explorer/explorer.tsx:150]: question, this looks to be from trace analytics. I see discover can also have interval per multiple of a time unit (e.g. 3 hours, 12 hours), is that possible in observability?

**Review by joshuali925 — APPROVED** [review]

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 3]
```diff
@@ -165,23 +169,25 @@ export const Explorer = ({
...
-        objectId: curIndex
-      }).catch((error: any) => {
-        console.log(`Unable to get saved timestamp for this index: ${error.message}`);
-      });
+      const savedTimestamps = await savedObjects
+        .fetchSavedObjects({
+          objectId: curIndex,
+        })
+        .catch((error: any) => {
+          console.log(`Unable to get saved timestamp for this index: ${error.message}`);
```
[the remark this answers is not in the document]
Comment (ps48) [reply dashboards-observability/public/components/explorer/explorer.tsx:189]: Not added in this PR but will change it 👍🏽

File: dashboards-observability/public/components/explorer/explorer.tsx (hunk 2 above)
Comment (ps48) [reply dashboards-observability/public/components/explorer/explorer.tsx:150]: This is inspired from trace-analytics 💯 . This code deals with the the interval options for the first request made to the PPL backend. Hence, it only uses time bucket from datepicker and assigns the best one to Auto.  

The dynamic interval part from discover, is populated post response (i.e. after the first response). For example, if the response buckets are in hours then the new interval option of 30 minutes is added. This part is not added to observability yet, we should add it later.

**Review by joshuali925 — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**